src/app.rb

```diff
@@ -0,0 +1,86 @@
+require_relative 'person'
+require_relative 'trimmer_decorator'
+require_relative 'capitalize_decorator'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  # list of books..................
+
+  def list_books
+    puts(@books.map { |book| "Title: \"#{book.title}\", Author \"#{book.author}\"" })
+  end
+
+  # list of people..............
+
+  def list_people
+    puts(@people.map { |person| "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" })
+  end
+  # create person.................
+
+  def create_person
+    print 'Do you want to create a Student (1) or a Teacher (2)? [Input the number]: '
+    option = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    case option
+    when '1'
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp.downcase
+      @people << Student.new(age, name, parent_permission: (permission == 'y'))
+    when '2'
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(specialization, age, name)
+    end
+    puts 'Person Created Successfully'
+  end
+
+  # create book..............
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book Created Successfully'
+  end
+  # create rental.....................
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.map.with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}" }
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not ID)'
+    @people.map.with_index do |person, index|
+      puts "#{index}) Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    print 'Date (YYYY/MM/DD): '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental Created Successfully'
+  end
```

Comment:
- [ ] As mentioned by the previous reviewer you should ensure that the display of the Create Rentals section matches the video reference, the person list should specify if a person is a student or teacher 🙃 

Your design

<img width="566" alt="Screenshot 2023-10-05 at 21 25 55" src="https://github.com/fatima-najafi/oop-school-library/assets/74501523/a18da504-ac85-4a7c-8239-f06aa39b468f">

Given design

<img width="566" alt="Screenshot 2023-10-05 at 22 03 42" src="https://github.com/fatima-najafi/oop-school-library/assets/74501523/42508940-cac0-4b72-93fe-f9dd2990612c">

Reply:
This change has not yet been implemented well

Reply:
- [ ] It will be great to ensure that when you are creating a rental, just before the name if the person, you display if the person is a student or a teacher. As illustrated in the previous review this is a major requirement for the UI. If you can check the image below, your application still does not show this ⬇️ 
![Screenshot from 2023-10-06 06-11-05](https://github.com/fatima-najafi/oop-school-library/assets/32361963/d5abdc90-2d69-4b48-97a0-5e708bb5698d)

Kindy fix this. Reach out to me if you are facing a challenge with this

Reply:
![Screenshot (516)](https://github.com/fatima-najafi/oop-school-library/assets/101437051/ef6247f8-c174-447d-bd88-d6562d68aef6)